Update to bevy 0.10

NiseVoid opened the pull request:
**The update is blocked by bevy-inspector-egui, but not really since it only affects examples and can easily be disabled**

To run examples: Disable inspector in the Cargo toml and the example code

I only ran examples and tests, they all seemed to work fine, but I'm not sure if any features have been missed in these. I did have some weird errors when closing windows, and some of the tests seemed to eat a lot of GPU resources, but I'm not sure if this is new behavior.

Some things to note:

- Shaders now need to specify global variables with `const` instead of `let`.
- You can no longer redefine a variable name with `let`.
- The set EffectSystems::ExtractEffects is unused, and only lives in the ExtractSchedule, so it can probably be removed
- There is a new field on `RenderPipelineDescriptor` called `push_constant_ranges`, I left it empty but there might be a better value.

djeedai commented:
Thanks for the preparatory work @NiseVoid. That's going to be very useful for a swift upgrade :) I'm going to wait the last minute to merge though, just in case there would be any need to release on v0.9 first (there was a lot of changes since v0.5). Better safe than reverted. But I do intend to release as soon as possible after Bevy v0.10 landed.

What I meant by "The update is blocked by bevy-inspector-egui" is that the examples (and therefore CI) can only pass once that crate upgraded. And v0.6 can only be published to crates.io after that, which is really what most folks are waiting for. I once did disable the dependency in examples and released without it, then re-enabled and released again; the result was a "gain" of 24h~48h earlier release over what would have happened if not disabled, but that gain was largely offset by the absolute pain of having to manually do 2 releases back-to-back, with all the overhead (including manual testing) involved. I wish that was not the case and I could just press a button, but because of the difficulty to test on GPU and lack of time to write comprehensive tests, we're unfortunately not quite there yet. So I'm unlikely to do that again, baring exceptional circumstances, and especially since Jakob is generally very diligent and timely with their update of `bevy-inspector-egui`.

janhohenheim commented:
This closes #147

djeedai commented:
@NiseVoid will you update that PR once `bevy-inspector-egui` is updated, or should I take over the upgrade? I don't mind either.

NiseVoid commented:
> @NiseVoid will you update that PR once `bevy-inspector-egui` is updated, or should I take over the upgrade? I don't mind either.

If that's the only issue with the PR, I guess it might be faster if you do that update and can then merge it when done.

I also tested this PR on my code, and it seems to behave identical to the commit it was based on.

djeedai commented:
> If that's the only issue with the PR, I guess it might be faster if you do that update and can then merge it when done.

That needs to be done as part of this PR though.

Also we should expect an updated `bevy-inspector-egui` for 0.10 within the hour.

NiseVoid commented:
Updated the inspector, rebased, and ran all examples (without any changes to the Cargo.toml or tests compared to what was comitted). Seems to all work fine
